stats/informationbox.cpp

```diff
@@ -11,74 +12,65 @@ static const int informationBorder = 2;
 static const double informationBorderRadius = 4.0; // Radius of rounded corners
 static const int distanceFromPointer = 10; // Distance to place box from mouse pointer or scatter item
 
-InformationBox::InformationBox() : RoundRectItem(informationBorderRadius, nullptr)
+InformationBox::InformationBox(StatsView &v) :
+	ChartRectItem(v, QPen(informationBorderColor, informationBorder),
+		      QBrush(informationColor), informationBorderRadius)
 {
-	setPen(QPen(informationBorderColor, informationBorder));
-	setBrush(informationColor);
-	setZValue(ZValues::informationBox);
 }
 
 void InformationBox::setText(const std::vector<QString> &text, QPointF pos)
 {
-	width = height = 0.0;
-	textItems.clear();
+	QFontMetrics fm(font);
+	double fontHeight = fm.height();
 
+	std::vector<double> widths;
+	widths.reserve(text.size());
+	width = 0.0;
 	for (const QString &s: text) {
-		if (!s.isEmpty())
-			addLine(s);
+		widths.push_back(static_cast<double>(fm.size(Qt::TextSingleLine, s).width()));
+		width = std::max(width, widths.back());
 	}
 
 	width += 4.0 * informationBorder;
-	height += 4.0 * informationBorder;
+	height = widths.size() * fontHeight + 4.0 * informationBorder;
+
+	ChartRectItem::resize(QSizeF(width, height));
 
-	// Setting the position will also set the proper size
-	setPos(pos);
+	painter->setPen(QPen(darkLabelColor)); // QPainter uses QPen to set text color!
+	double y = 2.0 * informationBorder;
+	for (size_t i = 0; i < widths.size(); ++i) {
+		QRectF rect(2.0 * informationBorder, y, widths[i], fontHeight);
+		painter->drawText(rect, text[i]);
```

Comment:
ah, so I misunderstood what this was actually doing. It's not actually doing OpenGL - it still uses the same drawing primitives, etc. I was confused there, clearly.
This makes a lot more sense.

Reply:
Actually it does use OpenGL (or whatever is the shader language du-jour).

I paints once onto a QImage with the usual QPainer API, then transforms this into a texture and that is blitted with OpenGL when the box is moved.

For the regression line, etc. I'm not yet sure if we also should use textures or directly use OpenGL (or whatever) primitives.

Reply:
But what I mean is that OpenGL is an implementation detail that we hopefully can rely on being handled in the background. We don't need to write OpenGL code. And this will still work on platforms that are abandoning OpenGL (as macOS is).
I really hope we will not be sucked into writing OpenGL code here - because there be dragons.